### ISSUE TITLE
Outputting monthly restart files

### ISSUE THREAD

**Issue by bsuther** [geoschem/geos-chem#825]
Hello,
Can you tell me how to make it output restart files at the start of each month during a longer simulation? I used to get it to do that by changing the "Restart.frequency:          'End',"  to   "Restart.frequency:          00000001 000000," as described [here](https://github.com/geoschem/geos-chem/issues/310) however, since I updated to version 13.1.0 it no longer seems to be working. I see that a similar issue was discussed for GCHP [in this issue](https://github.com/geoschem/geos-chem/issues/441), is this the case even with non-GCHP runs? I am trying to use version 13.1.0, using the out of the box code with RRTMG turned on and boundary conditions. 
Thank you!

**Comment by lizziel** [comment]
Hi @bsuther, as far as I know the frequency functionality for the Restart collection should still work with GEOS-Chem Classic using the format you specific. Doing the same thing for GCHP is not possible, though the behavior can be achieved starting in 13.1 by specifying exact dates to output. Since this issue is about GC-Classic, and that hack is for GCHP only, I won't go into more specifics about that.

We can try to replicate the GC-Classic issue you are seeing on our end and will get back to you.

**Comment by lizziel** [comment]
Hi @bsuther, I have not been able to replicate this issue with our benchmark simulation in 13.1. Could you give more specifics on your test, including your `input.geos`, `HISTORY.rc,` and run log files? If you append `.txt` to the end of the filenames you can drag and drop them into the comment box.

**Comment by bsuther** [comment]
Hi @lizziel thank you for working on this. Attached are the files. It did create one Restart File, but not at the time I expected (GEOSChem.Restart.20140101_0010z.nc4). 
[HISTORY.rc.txt](https://github.com/geoschem/geos-chem/files/7009609/HISTORY.rc.txt)
[input.geos.txt](https://github.com/geoschem/geos-chem/files/7009610/input.geos.txt)
[GC.log](https://github.com/geoschem/geos-chem/files/7009611/GC.log)
[HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/7009613/HEMCO_Config.rc.txt)

**Comment by lizziel** [comment]
I think I see your problem. Your Restart collection frequency is set to daily but your duration is still End. This means you will output daily Restart data but to one file only. If you change duration to match frequency then you will get one time per file.

I tested out the case similar to your run to see whap would happen, using a 3-hr run with Restart frequency set to 1-hr and duration set to End. I got two files output:
```
GEOSChem.Restart.20190701_0010z.nc4
GEOSChem.Restart.20190701_0300z.nc4
```
The first file, timestamped 10 minutes into the run, contains two times (50 and 110) with units "minutes since 2019-07-01 00:10:00". The second file, timestamped 3 hours into the run, contains one time (0) with units "minutes since 2019-07-01 03:00:00". So current behavior for the case of duration End and frequency less than duration is to output two files: one at the end and one with all other intervals leading up to end.

This behavior is a little odd, particularly the times in the file with mid-run restart values. The filename string is one timestep into the run (GEOSChem.Restart.20140101_0010z.nc4). This would be very confusing to use since it doesn't contain data for that time. @yantosca, is this expected behavior? I wonder if we need to more thoroughly test different combinations of frequency and duration in GC-Classic History. For the case of the Restart collection maybe we should force duration and frequency to match, and error out if they do not.

**Comment by yantosca** [comment]
@bsuther @lizziel, I can push a fix for this into the patch/13.2.1 branch, which will ship with 13.2.1.  It will be very easy to force an error if freqeuncy & duration don't match for the Restart collection.

**Comment by yantosca** [comment]
I have pushed commit 7648b19a605cda15be1e35942a0522f30fa46014 that will resolve this issue.  This will ship with 13.2.1.

NOTE: if you set, e.g.
```
in input.geos:
   Start YYYYMMDD, hhmmss  : 20190101 000000
   End   YYYYMMDD, hhmmss  : 20190102 000000

in HISTORY.rc:
  Restart.frequency:          00000000 030000,
  Restart.duration:           'End',
```
then you will get an error message such as this, because the duration (1 day) doesn't match the frequency (3 hrs).
```
===============================================================================
GEOS-Chem ERROR: Incompatible Restart collection metadata!           
Restart.frequency = 00000000 030000 but Restart.duration  = 00000001 000000
 -> at History_ReadCollectionData (in module History/history_mod.F90)
===============================================================================
```
But if you set these:
```
```
in input.geos:
   Start YYYYMMDD, hhmmss  : 20190101 000000
   End   YYYYMMDD, hhmmss  : 20190102 000000

in HISTORY.rc:
  Restart.frequency:          00000001 000000,
  Restart.duration:           'End',
```
then the run should proceed normally, because "End" is interpreted as 1 day, based on the start & end time of the run in input.geos.

In other words, we don't simply test that the string "000000001 000000" /= "End", but we test if the `FileWriteYmd /= FileCloseYmd` and 'FileWriteHms /= FileCloseHms` for the Restart collection.

**Comment by yantosca** [comment]
I can close out this issue now.  We are currently waiting for 13.2.0 to pass benchmarks, then 13.2.1 will be released immediately afterwards.

**Comment by bsuther** [comment]
I made the suggested changes to HISTORY.rc and it is working as I expected now. Thank you both for your help!
